plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -287,6 +287,9 @@ public static Object convert(Object value, Type type)
             // TODO the client sees the bytearray's tostring ouput instead of seeing actual bytes, needs to be fixed.
             return ((ByteBuffer) value).array();
         }
+        if (type instanceof Types.DateType) {
+            return Long.parseLong(value.toString());
```

Comment:
I have tried to add a branch for DateType, but it still goes to the branch for TimestampType.
What shall I do for this? @hashhar